libs/use-multi-media-devices/src/index.ts

```diff
@@ -0,0 +1,474 @@
+import { useState, useEffect, useReducer } from 'react';
+
+export type MediaDevices = {
+  cameraList: InputDeviceInfo[];
+  microphoneList: InputDeviceInfo[];
+  streams: StreamsMap;
+  initDefaultStream: () => void;
+  addStream: ({
+    type,
+    microphone,
+    camera,
+    audioConstraints,
+    videoConstraints,
+  }: {
+    type: StreamTypes;
+    microphone?: InputDeviceInfo;
+    camera?: InputDeviceInfo;
+    audioConstraints?: MediaTrackConstraints;
+    videoConstraints?: MediaTrackConstraints;
+  }) => Promise<void>;
+  toggleAudio: (id: StreamId) => void;
+  toggleVideo: (id: StreamId) => void;
+  removeStream: (id: StreamId) => void;
+  applyConstraints: ({
+    id,
+    audioConstraints,
+    videoConstraints,
+  }: {
+    id: StreamId;
+    audioConstraints?: MediaTrackConstraints;
+    videoConstraints?: MediaTrackConstraints;
+  }) => Promise<void>;
+  reset: () => void;
+};
+
+export type Resolution = {
+  name?: string;
+  width: number;
+  height: number;
+};
+
+const allResolutions: Resolution[] = [
+  {
+    name: '4K',
+    width: 3840,
+    height: 2160,
+  },
+  {
+    name: 'QHD',
+    width: 2560,
+    height: 1440,
+  },
+  {
+    name: 'FHD',
+    width: 1920,
+    height: 1080,
+  },
+  {
+    name: 'HD',
+    width: 1280,
+    height: 720,
+  },
+  {
+    name: '480p',
+    width: 854,
+    height: 480,
+  },
+  {
+    name: 'SD',
+    width: 640,
+    height: 480,
+  },
+  {
+    name: '360p',
+    width: 640,
+    height: 360,
+  },
+];
```

Comment:
Similar suggestion to https://github.com/dolbyio-samples/rts-app-react-publisher-viewer/pull/190/files#r1040384622, but with constants in a colocated `./constants.ts` file.

Reply:
Same as above